src/site/content/en/fast/efficiently-load-third-party-javascript/index.md

```diff
@@ -0,0 +1,174 @@
+description: |
+  Learn how to avoid common pitfalls of using third-party scripts and improve load times and user experience.
+hero: hero.jpg
+alt: Aerial view of shipping containers.  
+tags:
+  - post # post is a required tag for the article to show up in the blog.
+  - performance
+  - third-party
+  - javascript
+---
+
+If a third-party script is [slowing down](/third-party-javascript/) your page load, you have two options to improve performance:
+
+* Remove it if it doesn't add clear value to your site.
+
+* Optimize the loading process.
+
+This guide explains how to optimize the loading process of third-party scripts with the following techniques: 
+
+1. Using the `async` or `defer` attribute on `<script>` tags
+
+2. Establishing early connections to required origins
+
+3. Lazy-loading
+
+4. Optimizing how you serve third-party scripts 
+
+## Use `async` or `defer`
+
+Because [synchronous scripts](link-to-intro) delay DOM construction and rendering, you should always load third-party scripts asynchronously unless the script has to run before the page can be rendered.
+
+The `async` and `defer` attributes tell the browser that it may go on parsing the HTML while loading the script in the background, and then execute the script after it loads. This way, script downloads don’t block DOM construction and page rendering. The result is that the user can see the page before all scripts have finished loading.
+
+```html
+<script async src="script.js">
+
+<script defer src="script.js">
+```
+
+The difference between `async` and `defer` is when they start executing the scripts.
+
+### `async`
+
+Scripts with the `async` attribute execute at the first opportunity after they finish downloading and before the window’s [load](https://developer.mozilla.org/en-US/docs/Web/Events/load) event. This means it’s possible (and likely) that `async` scripts will not be executed in the order in which they appear in the HTML. It also means they can interrupt DOM building if they finish downloading while the parser is still at work.
+
+![Diagram of parser blocking script with async attribute](async@2x.png)
+
+### `defer`
+
+Scripts with the `defer` attribute execute after HTML parsing is completely finished, but before the [`DOMContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/Events/DOMContentLoaded) event. `defer` guarantees scripts will be executed in the order they appear in the HTML and will not block the parser.
+
+![Diagram of parser flow with a script with defer attribute](defer@2x.png)
+
+* Use `async` if it's important to have the script run earlier in the loading process. 
+
+* Use `defer` for less critical resources. A video player that's below-the-fold, for example.
+
+Using these attributes can significantly speed up page load. For example, [Telegraph recently deferred all of their scripts](https://medium.com/p/a0a1000be5#4123), including ads and analytics, and improved the First Ad Loaded metric by an average of four seconds.
```

Comment:
Is First Ad Loaded a metric that Telegraph devised? (I didn't find anything other than the Telegraph Medium post with a quick Google search.) If so, it seems weird to present it as proper noun?